index.php

```diff
@@ -1,15 +1,3 @@
 <?php
-
-define('ROOT_DIR', realpath(dirname(__FILE__)) . '/');
```

Comment:
The `ROOT_DIR`, `LIB_DIR` and `VENDOR_DIR` constants are required before calling the constructor, so it's impossible to move them there.

Splitting constant definition (`global.php`) and instantiation of Pico allows developers to integrate Pico into their own ecosystem and overwrite the constants as necessary. It allows e.g. the following:
```php
<?php

class MyApplication {
    protected $pico;

    // snip

    protected function initPico() {
        // either include the global.php
        require_once(PICO_DIR . 'global.php');
        // or define the constants on your own

        $this->pico = new Pico();
    }
    
    // snip
}
```

Anyway, this is no good solution at all, I focused too much on the existing code. I'll overhaul this, probably by replacing the constants with something better... Thanks for bringing this to my attention.